google-cloud-spanner/src/main/java/com/google/cloud/spanner/AsyncResultSetImpl.java

```diff
@@ -275,7 +282,7 @@ public void run() {
               switch (response) {
                 case DONE:
                   state = State.DONE;
-                  closeDelegateResultSet();
```

Comment:
The delegate `ResultSet` should not be closed by the callback executor, but only by the executor that fetches and buffers the rows (in the `ProduceRowsCallable` at line 380)